tensorflow/core/nccl/nccl_manager.cc

```diff
@@ -16,15 +16,32 @@ limitations under the License.
 
 #include <utility>
 
-#ifdef GOOGLE_CUDA
+#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
 
 #include "tensorflow/core/lib/core/refcount.h"
 #include "tensorflow/core/lib/core/threadpool.h"
-#include "tensorflow/core/platform/cuda.h"
 #include "tensorflow/core/platform/env.h"
+#if GOOGLE_CUDA
+#include "tensorflow/core/platform/cuda.h"
+#elif TENSORFLOW_USE_ROCM
+#include "tensorflow/core/platform/rocm.h"
+#endif
 
 namespace tensorflow {
 
+#if GOOGLE_CUDA
+using se::cuda::ScopedActivateExecutorContext;
+#elif TENSORFLOW_USE_ROCM
+using se::rocm::ScopedActivateExecutorContext;
+// Local hipify of cuda symbols
+#define cudaError_t hipError_t
```

Comment:
@jeffdaily would it be possible to include "core/util/gpu_kernel_helper.h" and use `gpuXXXX` for these types?

Reply:
As soon as I include `tensorflow/core/util/gpu_kernel_helper.h`, I get errors such as 
```
In file included from ./tensorflow/core/util/gpu_kernel_helper.h:25:0,
                 from tensorflow/core/nccl/nccl_manager.cc:29:
./tensorflow/core/util/gpu_device_functions.h: In function 'tensorflow::detail::GpuGridRange<T> tensorflow::GpuGridRangeX(T)':
./tensorflow/core/util/gpu_device_functions.h:110:34: error: 'blockIdx' was not declared in this scope
   return detail::GpuGridRange<T>(blockIdx.x * blockDim.x + threadIdx.x,
                                  ^
```

Also, `gpu_kernel_helper.h` does not cover every symbol that I need to locally hipify.  Those are `hipGetDevice`, `hipSetDevice`, and `hipSuccess`.

Is `using` preferred over `#define`?

Reply:
Including `gpu_kernel_helper.h` would incur a lot of other HIP/CUDA dependencies so let's ignore this for now.

And yes `using` is more preferred over `#define`. I could imagine these `#define` would be frowned upon when we file them upstream.

Reply:
Then again, `using` is only for aliasing types, not the functions I mentioned earlier.  Is there any other, better alternative for the `#define` statements?  I was trying to avoid duplicating the file only for the small hipXXX changes.

Reply:
in areas other than NCCL you could spot we did a lot of changes to merge `cuXXX` and `hipXXX` to `gpuXXX`. But all of those are encapsulated inside `gpu_kernel_helper.h` which is not suitable be included here.... so maybe let's just leave these `#define` as-is. We'll face the music upstreaming.